### DISCUSSION TITLE
Use custom JDK

### DISCUSSION THREAD

**Discussion by thumannw** [JetBrains/Qodana#158] [Q&A] [answered]
From the [docs](https://www.jetbrains.com/help/qodana/configure-jdk.html#Mount+JDK) I assume that putting a JDK into the folder `/root/.jdks/jdk` makes Qodana use that JDK for scanning, correct? I tried it but it didn't seem to work. It was an Amazon Corretto 1.8. Maybe 1.8 is not detected correctly? Setting JAVA_HOME to that (or any other location) seems to work. Was it meant to be used this way?

**Comment by brichbash** [comment, marked as answer]
Hello,

> makes Qodana use that JDK for scanning, correct?

It does not, actually. We should supplement the documentation with that information. Thanks for bringing this up.

If you want to use Corretto 1.8, you might as well use `projectJDK` property in `qodana.yaml`.
For example:
```
version: "1.0"
linter: jetbrains/qodana-jvm:2022.3-eap
projectJDK: corretto-1.8
```
Here, `corretto-1.8` is `suggested_sdk_name` from https://download-cdn.jetbrains.com/jdk/feed/v1/jdks.json.xz. You can use any other JDKs compatible with linux from the link, and Qodana will download it automatically.

**Reply by thumannw** [reply]
In our case, the bootstrapping of the JDK is baked into Gradle, and the build setup demands that exactly this distribution is used when running Gradle. So my idea was to create a link at `/root/jdks/.jdk` pointing to this distribution.
Even if one could use public distributions, it is often a company policy that build systems do not communicate with the outside internet.
So, I guess setting `JAVA_HOME` is the only way to use a custom JDK for the scan?

**Reply by brichbash** [reply]
In that case, yes. It is the only way at the moment. If you have any ideas on how this use case might be covered, please create a feature request on [our issue tracker](https://youtrack.jetbrains.com/issues?q=project:%20Qodana). We would be very grateful.
